bigquery/google/cloud/bigquery/job.py

```diff
@@ -2475,237 +2475,268 @@ def __eq__(self, other):
-    """Map a single entry in a query plan.
+    """QueryPlanEntry represents a single stage of a query execution plan.
 
-    :type name: str
-    :param name: name of the entry
+    See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs
+    for the underlying API representation within query statistics.
 
-    :type entry_id: int
-    :param entry_id: ID of the entry
-
-    :type start_ms: int
-    :param start_ms: start time for this stage of the query plan
-
-    :type end_ms: int
-    :param end_ms: end time for this stage of the query plan
+    """
 
-    :type input_stages: List(int)
-    :param input_stages: list of stage IDs that are inputs for this stage
+    _PROPERTY_TO_API_FIELD = {
+        'name': 'name',
+        'entry_id': 'id',
+        'start': 'startMs',
+        'end': 'endMs',
+        'input_stages': 'inputStages',
+        'parallel_inputs': 'parallelInputs',
+        'completed_parallel_inputs': 'completedParallelInputs',
+
+        'wait_ms_avg': 'waitMsAvg',
+        'wait_ms_max': 'waitMsMax',
+        'wait_ratio_avg': 'waitRatioAvg',
+        'wait_ratio_max': 'waitRatioMax',
+
+        'read_ms_avg': 'readMsAvg',
+        'read_ms_max': 'readMsMax',
+        'read_ratio_avg': 'readRatioAvg',
+        'read_ratio_max': 'readRatioMax',
+
+        'compute_ms_avg': 'computeMsAvg',
+        'compute_ms_max': 'computeMsMax',
+        'compute_ratio_avg': 'computeRatioAvg',
+        'compute_ratio_max': 'computeRatioMax',
+
+        'write_ms_avg': 'writeMsAvg',
+        'write_ms_max': 'writeMsMax',
+        'write_ratio_avg': 'writeRatioAvg',
+        'write_ratio_max': 'writeRatioMax',
+        'records_read': 'recordsRead',
+        'records_written': 'recordsWritten',
+        'status': 'status',
+        'shuffle_output_bytes': 'shuffleOutputBytes',
+        'shuffle_output_bytes_spilled': 'shuffleOutputBytesSpilled',
+        'steps': 'steps',
+     }
 
-    :type parallel_inputs: int
-    :param completed_parallel_inputs: number of parallel input
-            segments in the stage
+    def __init__(self):
+        self._properties = {}
 
-    :type completed_parallel_inputs: int
-    :param completed_parallel_inputs: number of parallel input
-            segments completed
+    @classmethod
+    def from_api_repr(cls, resource):
+        """Factory: construct instance from the JSON repr.
 
-    :type wait_ms_avg: int
-    :param wait_ms_avg: milliseconds the average shard spent
-            waiting to be scheduled
+        Args:
+            resource(Dict[str: object]):
+                ExplainQueryStage representation returned from API
 
-    :type wait_ms_max: int
-    :param wait_ms_max: milliseconds the slowest shard spent
-            waiting to be scheduled
+        Returns:
+            google.cloud.bigquery.QueryPlanEntry:
+                Query plan entry parsed from ``resource``
+        """
+        entry = cls()
+        entry._properties = copy.deepcopy(resource)
```

Comment:
Probably not worth doing the copy for a read-only class. Plus @alixhami and I are thinking about dropping this copy everywhere so that it's possible to do things like `job_config.destination_time_partitioning.field = 'field_name'` (we already dropped it from `TimePartitioning` for this example).

Reply:
Will switch to direct assignment to _properties then.